src/geometry/PolySet.h

```diff
@@ -8,12 +8,20 @@
 
 #include <vector>
 #include <string>
+#include <unordered_map>
+#include <hash.h>
+
+int operator==(const Vector3d &a, const Vector3d &b);
 
 class PolySet : public Geometry
 {
 public:
   VISITABLE_GEOMETRY();
-  Polygons polygons;
+  PolygonsInd polygons_ind;
+  std::vector<Vector3d> points;
+  std::unordered_map<Vector3d, int> pointMap;
```

Comment:
This adds unnecessary weight to the PolySet class, and should be part of a `PolySetBuilder` or similar.